### DISCUSSION TITLE
Collections inside of collections aren't a thing? What should I do instead?

### DISCUSSION THREAD

**Discussion by samfcheng** [pocketbase/pocketbase#1586] [Q&A]
So, collections inside of collections aren't a thing, right?

Here's an example of what I'm trying to do with a link in bio/mini site creator.
Each user has sites, and each site has a variety of buttons on it.

Ideally, the structure would look something like this:
- _sites_ (collection)
  - examplesite1 (record)
     -  user (relation)
     - title
     - _buttons_ (collection)
       - button1 (record)
         - text
         - url
       - button2 (record)
         - text
         - url

Since I can't do this in pocketbase, how can I achieve this same function?

I don't want to create button collections for each site and reference it back to the original site collection because it's too messy.

**Comment by ganigeorgiev** [comment]
_This was discussed in https://github.com/pocketbase/pocketbase/discussions/1536, https://github.com/pocketbase/pocketbase/discussions/1152, and probably others._

In short, the only option to represent a structured relational data at the moment is to have separate collections and associate them with a `relation` field.
Alternatively, you can have a `json` field and load it with whatever data you want, but there are no validations for it out of the box.

_Note that the relation management UI will be improved in the near future as part of #976 and the other planned Admin UI changes for v0.12.0._
